ec/ecdatasource/deploymentsdatasource/schema.go

```diff
@@ -0,0 +1,118 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package deploymentsdatasource
+
+import "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+func newSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name_prefix": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"healthy": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"deployment_template_id": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"return_count": {
+			Type:     schema.TypeInt,
+			Computed: true,
+		},
+		"deployments": {
+			Type:     schema.TypeList,
+			Computed: true,
+			Elem:     newDeploymentList(),
+		},
+
+		// Deployment resources
+		"elasticsearch": {
+			Type:     schema.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem:     newResourceFilters(),
+		},
+		"kibana": {
+			Type:     schema.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem:     newResourceFilters(),
+		},
+		"apm": {
+			Type:     schema.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem:     newResourceFilters(),
+		},
+		"enterprise_search": {
+			Type:     schema.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem:     newResourceFilters(),
+		},
+	}
+}
+
+func newDeploymentList() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
```

Comment:
Is it intentional to leave the model so bare? I thought we were going to map each of the found deployments with the same attributes as `ec_deployment`

Reply:
Yes, this was intentional. That's one of the reasons I divided #8 into two datasources. This way one of the datasources (ec_deployment) would give you detailed information about a single resource and this datasource would only give you deployment IDs which you can later use for batch actions. Similar to https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/autoscaling_groups

But I'm not attached to my solution, if you disagree with me or have some other use case in mind, happy to change it! :)

Reply:
I guess that makes sense, I'm not opposed to it at all 👍